cpp/include/raft/neighbors/ivf_list.hpp

```diff
@@ -67,9 +67,9 @@ list<SpecT, IdxT, SizeT>::list(raft::device_resources const& res,
  * Resize a list by the given id, so that it can contain the given number of records;
  * copy the data if necessary.
  */
-template <template <typename> typename SpecT, typename IdxT, typename SizeT>
+template <template <typename> typename SpecT, typename ValueT, typename IdxT, typename SizeT>
```

Comment:
Why is the need to add the `ValueT`? Is `SpecT<SizeT>::value_type` not enough for something?

Reply:
`SpecT<SizeT>::value_type` is not enough, to change the value_type I would have to use a template on Spec, and I think it makes more sense to have the value type attached to the list than to the Spec.

Reply:
I reverted this change to bring back `SpecT::value_type`. For `IdxT` I don't mind keeping it like that but if you would prefer to see it in `SpecT` I can make the change.